### PR TITLE
701 empty records

### LINKED ISSUE #701: Update no age group section
Updated this to simplify. 
Screen Shot 2020-11-09 at 3 25 05 PM

- Change no age-group section to "incomplete enrollments". 
- Deleting the child record if the user navigates away before finishing, but to preserve the data if the user is logged out or needs to refresh the page. 

Not for this ticket: We can revisit how we're handling this in the future to avoid unintentionally created incomplete enrollments (a confirmation modal when you navigate away asking if you want to save the in-progress enrollment? )

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #701

### CODE REVIEW

**Comment by jlhogan** [comment]
gonna be honest,  dont love the extent to which this totally bucks a lot of the patterns we've been following so far and also some standard HTTP patterns, but if others feel good about this as the best/quickest way to move forward i wont object

**Comment by mmazanec22** [comment]
Yeah this felt hacky @jlhogan but I'm not sure what a better way forward is.  Suggestions welcome!

**Comment by mmazanec22** [comment]
Migrating comment from slack: "what about instead of still getting the successful 201, you get back a 400 with the child with validation errors as the body ?"

File: client/src/components/Forms/ChildIdentifiers/Form.tsx [hunk 1]
```diff
@@ -41,27 +42,55 @@ export const ChildIdentifiersForm = ({
...
+      apiPost('children', _child, {
+        accessToken,
       })
-      .finally(() => (isMounted() ? setSaving(false) : null));
+        .then((res) => {
+          history.replace({ pathname: `/create-record/${res.id}` });
+          afterSaveSuccess();
+        })
+        .catch((err) => {
+          console.log(err);
```
Comment (jlhogan) [inline client/src/components/Forms/ChildIdentifiers/Form.tsx:79]: dupe console log with console.err below

File: src/routes/children.ts [hunk 2]
```diff
@@ -101,7 +101,8 @@ childrenRouter.post(
   passAsyncError(async (req, res) => {
     try {
       const child = await controller.createChild(req.body, req.user);
-      res.status(201).send({ id: child.id });
+      // Send child because we need access to the validation errors
+      res.status(201).send(child);
```
Comment (jlhogan) [inline src/routes/children.ts:105]: this should change back right ?

**Review by jlhogan — APPROVED** [review]
few nits but overall LGTM

**Review by bamader — APPROVED** [review]
Nice! This all looks good (if there are still breaks in how this meshes with HTTP patterns, I confess ignorance on that front--this seems like just a good fix to me).

**Pull request merged**